src/main/java/com/blackducksoftware/integration/jira/task/maintenance/CleanUpOrphanedTicketsTask.java

```diff
@@ -0,0 +1,225 @@
+        } catch (final Exception e) {
+            logger.warn("There was a problem while attempting to clean up orphan tickets: " + e.getMessage());
+            return ERROR_STATUS_MESSAGE;
+        }
+        return DEFAULT_STATUS_MESSAGE;
+    }
+
+    private void findAndUpdateIssuesInBatches(final JiraIssueServiceWrapper issueServiceWrapper, final ApplicationUser user, final Query query, final CustomField projectVersionUrlField, final BlackDuckService blackDuckService)
+        throws JiraIssueException, IntegrationException {
+        int batchCount = 0;
+        int offset = 0;
+        int limit = MAX_BATCH_SIZE;
+        List<Issue> foundIssues;
+        do {
+            foundIssues = issueServiceWrapper.queryForIssues(user, query, offset, limit);
+            offset += limit;
+            processBatch(issueServiceWrapper, foundIssues, projectVersionUrlField, blackDuckService);
+            batchCount++;
+        } while (foundIssues.size() == limit || batchCount <= MAX_BATCHES_PER_RUN);
+    }
+
+    private void processBatch(final JiraIssueServiceWrapper issueServiceWrapper, final List<Issue> issuesToProcess, final CustomField projectVersionUrlField, final BlackDuckService blackDuckService)
+        throws IntegrationException, JiraIssueException {
+        for (final Issue issue : issuesToProcess) {
+            final String projectVersionUrl = (String) issue.getCustomFieldValue(projectVersionUrlField);
+            if (StringUtils.isNotBlank(projectVersionUrl)) {
+                try {
+                    blackDuckService.get(projectVersionUrl);
+                } catch (final BlackDuckApiException apiEception) {
+                    final IntegrationRestException restException = apiEception.getOriginalIntegrationRestException();
+                    final int statusCode = restException.getHttpStatusCode();
+                    if (404 == statusCode) {
+                        logger.debug("The Black Duck project version for " + issue.getKey() + " was not found on the current Black Duck server.");
+                        resolveIssue(issueServiceWrapper, issue);
+                    }
+                }
+            }
+        }
+    }
+
+    private void resolveIssue(final JiraIssueServiceWrapper issueServiceWrapper, final Issue issue) throws JiraIssueException {
+        final JiraWorkflow issueWorkflow = issueServiceWrapper.getWorkflow(issue);
+        if (issueWorkflow == null) {
+            logger.debug("Unknown workflow. No action will be taken.");
+            return;
+        }
+
+        if (!WorkflowHelper.matchesBlackDuckWorkflowName(issueWorkflow.getName())) {
+            logger.debug("This issue does not use the Black Duck plugin workflow. No action will be taken.");
+            return;
+        }
+
+        logger.debug("Resolving issue: " + issue.getKey());
+        final String workflowStepResolveName = BlackDuckJiraConstants.BLACKDUCK_WORKFLOW_TRANSITION_REMOVE_OR_OVERRIDE;
```

Comment:
Why not just use the constant instead of storing it in a variable?

Reply:
The constant is weirdly named, so I thought this might make it more clear what this is. The method name might make it clear enough though.